### DISCUSSION TITLE
Batching too many queries together results in 404 error

### DISCUSSION THREAD

**Discussion by IgnisDa** [trpc/trpc#1792] [Q&A] [answered]
If I have this query, then I get a 404 (NOT FOUND) error. 
![image](https://user-images.githubusercontent.com/60938164/163556873-fda52b23-0123-4cde-a807-0f220b777147.png)

However if I split the same query like so, then it works as expected.
![image](https://user-images.githubusercontent.com/60938164/163556927-10d6acad-f39f-471b-9c3f-2d4a2e582ec2.png)

I believe this is a bug in `@trpc/server`.

**Comment by KATT** [comment, marked as answer]
Maybe this will help? https://trpc.io/docs/links#setting-a-maximum-batch-size

The max uri length depends on your deployment environment so you'll have to tweak it accordingly

**Reply by KATT** [reply]
Seems odd you need it for that few number of queries though.

**Reply by IgnisDa** [reply]
@KATT I am using it a non nextjs environment. What is the corresponding setting in the vanilla client?

**Reply by KATT** [reply]
It depends on the server and the client calling, it's hard to answer as a general rule.

**Reply by KATT** [reply]
@jlalmes is working on improvements for next version where you can define a max uri length rather than max number of queries
